### PR TITLE
Migrate to use azure-servicebus-jms

### PR DESCRIPTION
**Pull request by Netyyyy**
# Description

migrate to use azure-servicebus-jms

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsAutoConfiguration.java [hunk 1]
```diff
@@ -60,19 +62,24 @@ AzureServiceBusJmsProperties serviceBusJmsProperties(AzureGlobalProperties azure
     }
 
     @Bean
-    @ConditionalOnExpression("'premium'.equalsIgnoreCase('${spring.jms.servicebus.pricing-tier}')")
-    ServiceBusJmsConnectionFactoryCustomizer amqpOpenPropertiesCustomizer(ObjectProvider<AzureServiceBusJmsCredentialSupplier> azureServiceBusJmsCredentialSupplier) {
+    @ConditionalOnExpression("'standard'.equalsIgnoreCase('${spring.jms.servicebus.pricing-tier}')")
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsAutoConfiguration.java:65]: Please add some comments here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsConnectionFactoryFactory.java [hunk 2]
```diff
@@ -39,42 +39,39 @@ <T extends ServiceBusJmsConnectionFactory> T createConnectionFactory(Class<T> fa
...
         try {
             T factory;
             if (properties.isPasswordlessEnabled()) {
-                String remoteUrl = String.format(AMQP_URI_FORMAT,
-                    properties.getNamespace() + "." + properties.getProfile().getEnvironment().getServiceBusDomainName(),
-                    properties.getIdleTimeout().toMillis());
-                factory = factoryClass.getConstructor(String.class).newInstance(remoteUrl);
+                String hostName =
+                    properties.getNamespace() + "." + properties.getProfile().getEnvironment().getServiceBusDomainName();
+                TokenCredentialProvider tokenCredentialProvider =
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsConnectionFactoryFactory.java:64]: nit: we have a problem here, users can't customize the token credential bean.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsPasswordlessConfiguration.java [hunk 3]
```diff
@@ -26,16 +24,4 @@ class ServiceBusJmsPasswordlessConfiguration {
     AzureServiceBusJmsCredentialSupplier azureServiceBusJmsCredentialSupplier(AzureServiceBusJmsProperties azureServiceBusJmsProperties) {
         return new AzureServiceBusJmsCredentialSupplier(azureServiceBusJmsProperties.toPasswordlessProperties());
     }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsPasswordlessConfiguration.java:26]: Do we still need this class?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsPasswordlessConfiguration.java:26]: And also in the factories or some other places.

**Review by saragluna — COMMENTED** [review]
- We should remove the passwordless configuration class
- We'd better test the standard tier (both password and passwordless
- We should test both password and passwordless for premium.

**Review by saragluna — APPROVED** [review]

**Comment by Netyyyy** [comment]
Hi @alzimmermsft ,please help approve, thanks

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**